test/ref_ops_test.cpp

```diff
@@ -6458,6 +6458,168 @@ TEST_CASE(quantizelinear)
+
+    // Create a dummy input to hold the random data
+    migraphx::shape input_fixed_shape1{migraphx::shape::float_type, {sample_size}};
+
+    migraphx::parameter_map params0;
+    params0["Input_1"] = migraphx::argument(input_fixed_shape1);
+
+    std::vector<uint64_t> seed_data = {seed};
+    params0["Seed"]                 = migraphx::argument(seed_shape, seed_data.data());
+    auto result                     = p.eval(params0).back();
+
+    std::vector<float> result_vec(sample_size);
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+
+    // Compare result with the STL's mt19937 generator
+    std::mt19937 gen(seed);
+    std::uniform_real_distribution<> dis(0.0, 1.0);
+    std::vector<float> rand_samples(sample_size);
+    std::generate(rand_samples.begin(), rand_samples.end(), [&]() { return dis(gen); });
+    EXPECT(migraphx::verify::verify_range(result_vec, rand_samples, 100000));
+}
+
+TEST_CASE(random_uniform_and_seed_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+
+    float control_seed = 0.0f;
+    size_t sample_size(20000);
+
+    //      Shape of the random data
+    migraphx::shape rs{migraphx::shape::float_type, {{1, 2}, {2, sample_size + 1}}};
+    auto input = mm->add_parameter("Input_1", rs);
+
+    // Runtime randomization seed
+    auto seed_input = mm->add_instruction(migraphx::make_op("random_seed"));
+    mm->add_instruction(migraphx::make_op("random_uniform"), seed_input, input);
+    p.compile(migraphx::make_target("ref"));
+
+    // Create a dummy input to hold the random data
+    migraphx::shape input_fixed_shape1{migraphx::shape::float_type, {sample_size}};
+
+    migraphx::parameter_map params0;
+    params0["Input_1"] = migraphx::argument(input_fixed_shape1);
+    auto result        = p.eval(params0).back();
+
+    std::vector<float> result_vec(sample_size);
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+
+    // Compare result with the STL's mt19937 generator.  Expected verify_range error ~0.4
+    std::mt19937 gen(control_seed);
+    std::uniform_real_distribution<> dis(0.0, 1.0);
+    std::vector<float> rand_samples(sample_size);
+    std::generate(rand_samples.begin(), rand_samples.end(), [&]() { return dis(gen); });
+    EXPECT(migraphx::verify::verify_range(result_vec, rand_samples, 1.e8));
+}
+
+TEST_CASE(random_seed_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    mm->add_instruction(migraphx::make_op("random_seed"));
+
+    p.compile(migraphx::make_target("ref"));
+    auto result = p.eval({}).back();
+    std::vector<uint32_t> result_vec1(1);
+    result.visit([&](auto output) { result_vec1.assign(output.begin(), output.end()); });
+    std::vector<uint64_t> result_vec2(1);
+
+    result = p.eval({}).back();
+    result.visit([&](auto output) { result_vec2.assign(output.begin(), output.end()); });
+    EXPECT(result_vec1[0] != result_vec2[0]);
```

Comment:
It looks like this is doing an implicit conversion from `uint32_t` to `uint64_t` and then comparing that the results are not the same between the two random number generations. Technically sequential numbers from a random number generation can be the same number. This might make the test fail randomly. I think all we can test is that we do get an integer value out of this function.

Reply:
I think the change in types is a mistake and I'm fixing it, but the real test here is that two identical calls to the random seed generator give different values.  We shouldn't drop the test for fear of a vanishingly unlikely false positive.  When testing random data, we have to be robust enough to deal with a certain amount of randomness.

Reply:
I don't think having the chance for a false negative is good to have in our unit tests. @umangyadav what is your opinion on this?

Reply:
Talked with Umang, keep it as is and add a note that this is not checking for randomness (would need to do a statistical analysis). And that this could possibly produce a false negative.

Reply:
Comment added